ProphecyTestCase.php

```diff
@@ -1,91 +1,74 @@
-<?php
+<?php declare(strict_types=1);
 
 namespace Prophecy\PhpUnit;
 
+use PHPUnit\Framework\AssertionFailedError;
+use PHPUnit\Framework\TestCase;
+use Prophecy\Exception\Doubler\DoubleException;
+use Prophecy\Exception\Doubler\InterfaceNotFoundException;
 use Prophecy\Exception\Prediction\PredictionException;
+use Prophecy\Prophecy\MethodProphecy;
+use Prophecy\Prophecy\ObjectProphecy;
 use Prophecy\Prophet;
 
-abstract class ProphecyTestCase extends \PHPUnit_Framework_TestCase
+abstract class ProphecyTestCase extends TestCase
 {
     /**
      * @var Prophet|null
      */
     private $prophet;
 
-    private $prophecyAssertionsCounted = false;
-
     /**
-     * @param string|null $classOrInterface
+     * @throws DoubleException
+     * @throws InterfaceNotFoundException
      *
-     * @return \Prophecy\Prophecy\ObjectProphecy
+     * @psalm-param class-string|null $type
      */
-    protected function prophesize($classOrInterface = null)
+    protected function prophesize(?string $classOrInterface = null): ObjectProphecy
     {
+        if (\is_string($classOrInterface)) {
+            $this->recordDoubledType($classOrInterface);
+        }
+
         return $this->getProphet()->prophesize($classOrInterface);
     }
 
-    protected function verifyMockObjects()
+    protected function verifyMockObjects(): void
     {
         parent::verifyMockObjects();
 
-        if (null === $this->prophet) {
+        if ($this->prophet === null) {
             return;
         }
 
         try {
             $this->prophet->checkPredictions();
-        } catch (\Exception $e) {
-            /** Intentionally left empty */
-        }
-
-        $this->countProphecyAssertions();
-
-        if (isset($e)) {
-            throw $e;
-        }
-    }
-
-    protected function tearDown()
-    {
-        if (null !== $this->prophet && !$this->prophecyAssertionsCounted) {
-            // Some Prophecy assertions may have been done in tests themselves even when a failure happened before checking mock objects.
```

Comment:
could this removal be the reason of the test that failed for you ?

Reply:
Good catch! Thanks.